src/rdb_protocol/terms/sort.cc

```diff
@@ -83,9 +83,9 @@ class orderby_term_t : public op_term_t {
         /* Add a sorting to the table if we're doing indexed sorting. */
         } else if (index.has()) {
             rcheck(tbl_slice.has(), base_exc_t::LOGIC,
-                   "Indexed order_by can only be performed on a TABLE or TABLE_SLICE.");
+                   "Indexed order_by can only be performed on a TABLE or TABLE_SLICE, not on a `NAME_OF_TYPE_ORDER_BY_WAS_CALLED_ON`.  Make sure order_by comes before any transformations (such as map) or filters.");
```

Comment:
These messages still contain the string `NAME_OF_TYPE_ORDER_BY_WAS_CALLED_ON`. What was your plan with that?
I think if you just remove the ", not on a `NAME_OF_TYPE_ORDER_BY_WAS_CALLED_ON`" part the change would be good.